src/sensors/aht10.rs

```diff
@@ -0,0 +1,155 @@
+// RustDuino : A generic HAL implementation for Arduino Boards in Rust
+// Copyright (C) 2021  Sanmati Pande, Indian Institute of Technology Kanpur
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>
+
+use crate::com::i2c;
+use crate::delay::delay_ms;
+use fixed_slice_vec::FixedSliceVec;
+
+pub struct AHT10<'a> {
+    address: u8,
+    i2c: i2c::Twi,
+    vec: FixedSliceVec<'a, u8>,
+}
+
+///Constant values for AHT10 temperature and humity sensor.
+const AHT10_INIT_CMD: u8 = 0xE1; //initialization command for AHT10/AHT15
+const AHT10_START_MEASURMENT_CMD: u8 = 0xAC; //start measurment command
+const AHT10_SOFT_RESET_CMD: u8 = 0xBA; //soft reset command
+const AHT10_INIT_CAL_ENABLE: u8 = 0x08; //load factory calibration coeff
+const AHT10_INIT_BUSY: u8 = 0x08; //Status bit for busy
+
+impl<'a> AHT10<'a> {
+    ///20ms delay to wake up.
+    ///I2C address is set.
+    ///Usage: rustduino::sensors::aht10::new()
+    pub fn new(&mut self) -> &'static mut Self {
+        delay_ms(20);
+
+        self.soft_reset();
+
+        if !self.initialise() {
+            unreachable!();
+        }
+        unsafe { &mut *(0x38 as *mut Self) }
+    }
+
+    ///Return pointer
+    pub fn get() -> &'static mut Self {
+        unsafe { &mut *(0x38 as *mut Self) }
+    }
+
+    ///Initiates the transmission by self initiating the sensor.
+    ///Returns true if done otherwise false.
+    ///Usage: rustduino::sensors::aht10::initialise()
+    pub fn initialise(&mut self) -> bool {
```

Comment:
There are some errors in usage I cannot call it like `rustduino::sensors::aht10::initalise`, it would be `ATH10::initialise. Also, since we added mod.rs, there aht10 is no longer present.